project_task_description_template/models/project_task.py

```diff
@@ -13,6 +13,14 @@ class ProjectTask(models.Model):
 
     @api.onchange("description_template_id")
     def _onchange_description_template_id(self):
-        if self.description_template_id:
-            description = self.description if self.description else ""
-            self.description = description + self.description_template_id.description
+        for task in self:
+            if not task.description_template_id:
+                continue
+            template_text = task.description_template_id.description or ""
+            if not template_text:
```

Comment:
empty templates should be disallowed on template creation. but good check here.
Add required = true on  description_template_id.description   and remove this.